src/router/router.go

```diff
@@ -0,0 +1,24 @@
+package router
+
+import (
+	"challenge/api/controller"
+	"challenge/config"
+	"challenge/middleware"
+
+	"github.com/gin-gonic/gin"
+)
+
+func Init() {
+	router := NewRouter()
+	router.Run(config.Appconfig.GetString("server.port"))
```

Comment:
Error return value of `router.Run` is not checked. 

```suggestion
	err := router.Run(config.Appconfig.GetString("server.port"))
	if err != nil {
		log.Fatalf("Failed to start server: %v", err)
	}
```

Any possible tool that could be helpful here?